src/main/scala/nl.knaw.dans.easy.deposit/authentication/EasyBasicAuthStrategy.scala

```diff
@@ -16,31 +16,58 @@
 package nl.knaw.dans.easy.deposit.authentication
 
 import javax.servlet.http.{ HttpServletRequest, HttpServletResponse }
+import nl.knaw.dans.easy.deposit.authentication.AuthUser.UserState
+import nl.knaw.dans.easy.deposit.logging._
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
-import org.scalatra.ScalatraBase
 import org.scalatra.auth.strategy.BasicAuthStrategy
+import org.scalatra.{ ScalatraBase, ServiceUnavailable, Unauthorized }
+
+import scala.util.{ Failure, Success }
 
 object EasyBasicAuthStrategy {}
 
-class EasyBasicAuthStrategy(protected override val app: ScalatraBase,
+class EasyBasicAuthStrategy(protected override val app: ScalatraBase with AbstractResponseLogger,
                             authenticationProvider: AuthenticationProvider,
                             realm: String
                            ) extends BasicAuthStrategy[AuthUser](app, realm)
   with DebugEnhancedLogging {
 
+  implicit val responseLogger: AbstractResponseLogger = app
+
   override def name: String = getClass.getSimpleName
 
   protected def validate(userName: String,
                          password: String)
                         (implicit request: HttpServletRequest,
                          response: HttpServletResponse
                         ): Option[AuthUser] = {
-    authenticationProvider
-      .authenticate(userName, password)
-      .getOrElse(None) // TODO or .getOrElse(halt(ServiceUnavailable.logResponse))
-    // or BadGateWay / InternalServer error depending on the type of error ?
+    def haltWithInvalidUser = {
+      app halt Unauthorized(body = "invalid username/password").logResponse
+    }
+
+    def haltWithRegisteredUser = {
+      app halt Unauthorized(body = "please confirm your registration before logging in").logResponse
```

Comment:
See the comment with screenshots above, this message could be improved

```suggestion
      app halt Unauthorized(body = "please confirm your registration first").logResponse
```